vignettes.Rmd

```diff
@@ -1,65 +1,85 @@
+In vignettes, more than anywhere else, the answer to "But how do I do X?" is often "Don't do X".
+
+[^vignettes-3]: The `inst/doc/` folder is where vignettes go once they're built, when `R CMD build` makes the package bundle.
+
+## Workflow for writing a vignette {#sec-vignettes-workflow-writing}
 
 To create your first vignette, run:
 
 ```{r, eval = FALSE}
 usethis::use_vignette("my-vignette")
 ```
 
-This will:
+This does the following:
+
+1.  Creates a `vignettes/` directory.
+
+2.  Adds the necessary dependencies to `DESCRIPTION`, i.e. adds knitr to the `VignetteBuilder` field and adds both knitr and rmarkdown to `Suggests`.
 
-1.  Create a `vignettes/` directory.
+3.  Drafts a vignette, `vignettes/my-vignette.Rmd`.
 
-2.  Add the necessary dependencies to `DESCRIPTION` (i.e. it adds knitr to the `Suggests` and `VignetteBuilder` fields).
+4.  Adds some patterns to `.gitignore` to ensure that files created as a side effect of previewing your vignettes are kept out of source control (we'll say more about this later).
 
-3.  Draft a vignette, `vignettes/my-vignette.Rmd`.
+This draft document has the the key elements of an R Markdown vignette and leaves you in a position to add your content.
+You also call `use_vignette()` to create your second and all subsequent vignettes; it will just skip any setup that's already been done.
 
-The draft vignette has been designed to remind you of the important parts of an R Markdown file.
-It serves as a useful reference when you're creating a new vignette.
-Once you have this file, the workflow is straightforward:
+Once you have the draft vignette, the workflow is straightforward:
 
-1.  Modify the vignette.
+1.  Start adding prose and code chunks to the vignette.
+    Use `devtools::load_all()` as needed and use your usual interactive workflow for developing the code chunks.
 
-2.  Press Ctrl/Cmd + Shift + K (or click ![](images/knit.png){width="45"}) to knit the vignette and preview the output.
+2.  Preview the rendered vignette periodically.
 
-3.  This builds with the installed package --- but you probably want the dev package.
-    Use `devtools::build_rmd()`.
+    If you're documenting parts of your package that are under active development, it is important to build the vignette with the *current* source package.
+    You need to ensure that the `library(yourpackage)` call in the setup chunk loads a bleeding edge version of your package.
+    Use `devtools::build_rmd("vignettes/my-vignette.Rmd")` in this case.
+    It builds your vignette against a (temporarily installed) development version of your package.
 
-The check workflow, `Cmd + Shift + E`, will run the code in all vignettes.
-This is a good way to verify that you've captured all the needed dependencies.
+    If it's OK to build the vignette using the most-recently-installed version of your package, it's fine to use your usual workflow for rendering `.Rmd` files.
```

Comment:
Cmd + Shift + B?

But I think this needs even more emphasis since it's a common source of confusion

Reply:
Added the advice to install the current source and generally added emphasis.